### PR TITLE
NIAD-648: adjust spine post + polling

### PR DESCRIPTION
**Pull request by bartek-sarul**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: docker/service/src/main/java/uk/nhs/adaptors/scr/models/ProcessingResult.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            .findFirst()
+            .orElseThrow(() -> {
+                LOGGER.error("Unable to extract HL7 from processing result response:\n{}", response);
+                return new ParseProcessingResultException("Unable to extract HL7 from processing result response");
+            })
+            .group();
+
+        return new ProcessingResult()
+            .setSoapEnvelope(soapEnvelope)
+            .setHl7(hl7);
```
Comment (dawidskainos) [inline docker/service/src/main/java/uk/nhs/adaptors/scr/models/ProcessingResult.java:40]: compilation is failing here

File: docker/service/src/main/java/uk/nhs/adaptors/scr/controllers/fhir/FhirController.java [hunk 2]
```diff
@@ -38,15 +41,23 @@
         produces = {APPLICATION_FHIR_JSON_VALUE, APPLICATION_FHIR_XML_VALUE})
     @ResponseStatus(HttpStatus.OK)
     public WebAsyncTask<ResponseEntity<?>> acceptFhir(
-        @RequestHeader("Content-Type") MediaType contentType, @RequestBody String body)
-        throws FhirValidationException {
+        @RequestHeader("Content-Type") @NotNull MediaType contentType,
+        @RequestHeader("Nhsd-Asid") @NotNull String nhsdAsid,
+        @RequestHeader("Party-Id-From") @NotNull String partyIdFrom,
```
Comment (dawidskainos) [inline docker/service/src/main/java/uk/nhs/adaptors/scr/controllers/fhir/FhirController.java:46]: I checked apigee setup - partyId is not passed as a header. Let's move it to application.yml for now - until we sort it out.

**Review by dawidskainos — CHANGES_REQUESTED** [review]

File: docker/README.md [hunk 3]
```diff
@@ -4,6 +4,25 @@ National Integration Adaptors - Summary Care Record
...
+Variables without a default value and not marked optional, *MUST* be defined for the adaptor to run.
+
+### General Configuration Options
+
+| Environment Variable               | Default                   | Description 
+| -----------------------------------|---------------------------|-------------
+| SCR_SERVER_PORT                    | 8080                      | The port on which the SCR API will run
+| SCR_LOGGING_LEVEL                  | INFO                      | Application logging level. One of: DEBUG, INFO, WARN, ERROR. The level DEBUG **MUST NOT** be used when handling live patient data.
+| SCR_PARTY_ID_TO                    |                           | Spine party id key
+| SCR_SPINE_URL                      |                           | Spine SCR URL
```
Comment (michael-kainos) [inline docker/README.md:21]: Where would someone get this URL from?

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -7,6 +7,8 @@ services:
       - "9000:8080"
     environment:
       - SCR_SPINE_URL=http://mock-spine-service:8081
+      - SCR_LOGGING_LEVEL
```
Comment (michael-kainos) [inline docker/docker-compose.yml:10]: Any where we can mention what is accepted as a value here?

File: docker/service/src/main/java/uk/nhs/adaptors/scr/models/ProcessingResult.java (hunk 1 above)
Comment (bartek-sarul) [reply docker/service/src/main/java/uk/nhs/adaptors/scr/models/ProcessingResult.java:40]: lombok.config was gitignored. All ok now

**Review by dawidskainos — DISMISSED** [review]

**Review by dawidskainos — APPROVED** [review]

**Pull request merged**